beacon/client/http-provider.go

```diff
@@ -387,12 +414,31 @@ func getRequestReader(ctx context.Context, requestPath string, providerAddress s
 	response, err := client.Do(req)
 	if err != nil {
 		// Remove the query for readability
-		trimmedPath, _, _ := strings.Cut(path, "?")
 		return nil, 0, fmt.Errorf("error running GET request to [%s]: %w", trimmedPath, err)
 	}
 	return response.Body, response.StatusCode, nil
 }
 
+// Log a request and add HTTP tracing to the context if the logger has it enabled
+func logRequest(ctx context.Context, methodName string, path string) context.Context {
```

Comment:
seems like this should be a receiver on BeaconHttpProvider so it can log info specific to the instance